src/ex4/server/services/autoDeleteCacheService.js

```diff
@@ -0,0 +1,34 @@
+const CACHE_FILE_AUTO_DELETE_TIME = 15000;
+
+const fs = require("fs");
+const path = require("path");
+const rimraf = require("rimraf");
+
+const cacheFileName = "cache.json";
+const itemFile = "./server/data/itemsList.json";
+const cacheDir = "./server/data/cache";
+const cacheFilePath = "./server/data/cache/cache.json";
+
+function autoDeleteCache() {
+  fs.readdir(cacheDir, function (err, files) {
+    files.forEach(function (file, index) {
+      fs.stat(path.join(cacheDir, cacheFileName), function (err, stat) {
+        if (err) {
+          return console.error(err);
+        }
+
+        setTimeout(() => {
```

Comment:
This is problematic code.
try and run your todo list and enter numbers one after the other like "1", "2", "3"....
The delete occurs randomly for no reason, each timeout you set here "lives" in the background and executes async resulting in cache being deleted for no real reason.

you should not handle cache file deletion by yourself, just make a smart data structure for cache, like:
cache.json:
`{
1: <timestamp>,
2: <timestamp>,
3: <timestamp>
}
`

then just check if current timestamp is 1min ahead of previous timestamp - if yes fetch again and overide, if no just take from the itemlist